src/linux/InotifyBackend.cc

```diff
@@ -165,7 +170,10 @@ bool InotifyBackend::handleSubscription(struct inotify_event *event, std::shared
     DirEntry *entry = sub->tree->add(path, CONVERT_TIME(st.st_mtim), S_ISDIR(st.st_mode));
 
     if (entry->isDir) {
-      watchDir(*watcher, entry, sub->tree);
+      bool success = watchDir(*watcher, entry, sub->tree);
+      if (!success) {
+        sub->tree->remove(path);
+      }
```

Comment:
I was not entirely sure if I should `return false` after this call. It looks like this drives the call to `watchers.insert((*it)->watcher);`.

Reply:
Pretty sure it should return false, but @devongovett probably knows better

Reply:
I added a `return false` now.